File: AFMobile/AFWinPhone/AFWinPhoneTests/UtilsTest.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using AFWinPhone.enums;
using AFWinPhone.rest.connection;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using AFWinPhone.utils;

namespace AFWinPhoneTests
{
    [TestClass]
    public class UtilsTest
    {
        [TestMethod]
        public void TestParseDate()
        {
            //first format
            DateTime? expected = DateTime.Parse("2013-01-06T04:00:00.000+0000");
            DateTime? actual = Utils.ParseDate("2013-01-06T04:00:00.000+0000");
            Assert.AreEqual(actual, expected);

            //second format
            expected = DateTime.ParseExact("22.02.2016", "dd.MM.yyyy", CultureInfo.InvariantCulture);
            actual = Utils.ParseDate("22.02.2016");
            Assert.AreEqual(actual, expected);

            //bad formats should not be parsed and return null
            actual = Utils.ParseDate("2011-01-18ijii");
            Assert.IsNull(actual);
        }

        [TestMethod]
        public void TestGetConnectionEndPoint()
        {
            AFSwinxConnection connection = new AFSwinxConnection("example.com", 8080, "/AFServer/resource", "http");

            String actual = Utils.GetConnectionEndPoint(connection);
            String expected = "http://example.com:8080/AFServer/resource";
            Assert.AreEqual(actual, expected);

            //test without port
            connection.setPort(0);
            expected = "http://example.com/AFServer/resource";
            actual = Utils.GetConnectionEndPoint(connection);
            Assert.AreEqual(actual, expected);
        }

        [TestMethod]
        public void TestValueOfSuccess()
        {
            SupportedWidgets expected = SupportedWidgets.TEXTFIELD;
            SupportedWidgets actual = Utils.ValueOf<SupportedWidgets>(typeof(SupportedWidgets), "TEXTFIELD");
            Assert.AreEqual(actual, expected);
        }

        [TestMethod]
        public void TestValueOfFail()
        {
            //test non existing value
            SupportedWidgets actual = Utils.ValueOf<SupportedWidgets>(typeof(SupportedWidgets), "non existing value");
            Assert.IsNull(actual);
            //test null value
            actual = Utils.ValueOf<SupportedWidgets>(typeof (SupportedWidgets), null);
            Assert.IsNull(actual);
        }

    }
}

File: AFMobile/AFAndroid/app/src/test/java/cz/cvut/fel/matyapav/afandroid/UtilsTest.java
package cz.cvut.fel.matyapav.afandroid;

import com.tomscz.afswinx.rest.connection.AFSwinxConnection;

import org.junit.Test;

import java.text.SimpleDateFormat;
import java.util.Date;

import cz.cvut.fel.matyapav.afandroid.utils.Utils;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

/**
 * Created by Pavel on 24.03.2016.
 */
public class UtilsTest {

    @Test
    public void testParseDate() throws Exception {
        //first format
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
        Date expected = format.parse("2013-01-06T04:00:00.000+0000");
        Date actual = Utils.parseDate("2013-01-06T04:00:00.000+0000");
        assertEquals(actual, expected);

        //second format
        format = new SimpleDateFormat("dd.MM.yyyy");
        expected = format.parse("22.02.2016");
        actual = Utils.parseDate("22.02.2016");
        assertEquals(actual, expected);

        //another formats should return null
        actual = Utils.parseDate("2011-01-18 00:00:00.0");
        assertNull(actual);
        System.out.println("Parsing date test PASSED");
    }

    @Test
    public void testGetConnectionEndPoint() throws Exception{
        AFSwinxConnection connection = new AFSwinxConnection("example.com", 8080, "/AFServer/resource", "http");

        String actual = Utils.getConnectionEndPoint(connection);
        String expected = "http://example.com:8080/AFServer/resource";
        assertEquals(actual, expected);

        //test without port
        connection.setPort(0);
        expected = "http://example.com/AFServer/resource";
        actual = Utils.getConnectionEndPoint(connection);
        assertEquals(actual, expected);
    }
}
